### DISCUSSION TITLE
Idea: Remote + Text Objects

### DISCUSSION THREAD

**Discussion by kevintraver** [folke/flash.nvim#325] [Ideas]
Given this example:

```python
def function(name):
    print(f"Hello, {name}!")

def function(input_list):
    for item in input_list:
        print(f"Item: {item}")

def function(dictionary):
    for key, value in dictionary.items():
        print(f"{key}: {value}")
```
Note: `<leader>s` is the keymap for this new mode (just to use as an example).

Typing `<leader>sir"` (mnemonic: _search inside remote quote_) would bring up all the quote _textobjects_ with a label:

```python
def function(name):
    print(f"Aello, {name}A") # <-- Labels

def function(input_list):
    for item in input_list:
        print(f"Btem: {itemB") # <-- Labels

def function(dictionary):
    for key, value in dictionary.items():
        print(f"Ckey}: {valueC") # <-- Labels
```

Selecting the label would then select the text object.

This would also work for operator mode.

Changing inside any quotes would only require typing `cri"[label]` (mnemonic: _change remote inside quote_)

Where as the the current treesitter way requires typing `cS[target][label]` (in this case `cSHe[label]`, where `S` is the _treesitter_search_ prefix, and `He[llo]` is the desired _jump_ destination)

This would be less cognitive overhead because it would not require trying to match the inside text for each example (which will often vary inside each quote) but instead just allow focusing on the desired text object.

The current treesearcher search is obviously very powerful, but this new way would be less keystrokes and more importantly an easier mnemonic. This would not be meant to replace treesitter search, but rather a companion or alternate to treesitter search (and it could still be powered by treesitter).

**Comment by kevintraver** [comment]
[mini.ai](https://github.com/echasnovski/mini.ai) does something similar with its **_next / last variants_**, but is obviously less useful for further away movements.

**Comment by julien-h2** [comment]
I was actually looking for a way to do a treesitter-search with only specific nodes! such as label all strings, label all functions, etc.

**Comment by kevintraver** [comment]
I now have a working implementation of this on my branch: https://github.com/kevintraver/flash.nvim/tree/feat/textobject
